### ISSUE TITLE
Docker file seems to flatten directory structure when COPYing files from subdirectories when pushing to a Balena device in local mode from Windows 10

### ISSUE THREAD

**Issue by TobiasEmil** [balena-io/balena-cli#1133]
- Balena version 9.14.6
- Standalone install: yes
- node -v v8.11.4
- npm -v 6.9.0
- Windows 10 x64

I am trying to compile my typescript in the Docker file, but when I COPY files from my dev computer to the Docker container it seems like the files of the subdirectories are being flattened into the workdir.

I have a very simple file structure as I have just initiated the project:

```
Dockerfile
start.sh
src/
- index.ts
- tsconfig.json
```
My Dockerfile looks as follows:

```
FROM resin/raspberrypi3-node:10.0.0
WORKDIR /usr/src/app
COPY package.json package.json
COPY package-lock.json package-lock.json
RUN npm i -g typescript
RUN JOBS=MAX npm install --production --unsafe-perm && npm cache verify && rm -rf /tmp/*

COPY . ./
RUN tsc -p src
CMD ["bash", "start.sh"]
```
 
When building and pushing the image to the Balena device in local mode I get the following error when the execution of the Dockerfile reaches the ‘RUN tsc -p src’ command:

```
[Build]   [main] Step 8/11 : RUN tsc -p src
[Build]   [main]  ---> Running in b843b59050da
[Build]   [main] error TS5058: The specified path does not exist: 'src'.
Some services failed to build:
        main: The command '/bin/sh -c tsc -p src' returned a non-zero code: 1
```

Likewise I cannot cd to the subdirectory and just run the tsc command from there:

```
[Build]   [main] Step 8/11 : RUN cd src
[Build]   [main]  ---> Running in 7620e8a516f3
[Build]   [main] /bin/sh: 1: cd: can't cd to src
```

I inserted a ‘RUN ls’ command into my Dockerfile before ‘RUN tsc -p src’ to verify the files are being copied which prints out the following:

```
[Build]   [main] Step 8/11 : RUN ls
[Build]   [main]  ---> Running in bf8b3c63454e
[Build]   [main] CHANGELOG.md
[Build]   Dockerfile
[Build]   README.md
[Build]   node_modules
[Build]   package-lock.json
[Build]   package.json
[Build]   src\index.ts
[Build]   src\tsconfig.json
[Build]   start.sh
[Build]   [main] Removing intermediate container bf8b3c63454e
[Build]   [main]  ---> 27ed6ea5eb9b
[Build]   [main] Step 9/11 : RUN tsc -p src
[Build]   [main]  ---> Running in 6ac5f86b5637
[Build]   [main] error TS5058: The specified path does not exist: 'src'.
```

And i am able to cat the files:

```
[Build] [main] Step 8/11 : RUN cat ‘src\tsconfig.json’
[Build] [main] —> Running in 5244084a805f
[Build] [main] {
[Build] “compilerOptions”: {
[Build] “module”: “commonjs”,
[Build] “target”: “es5”,
[Build] “sourceMap”: true,
[Build] “removeComments”: true,
[Build] “noImplicitAny”: true,
[Build] “outDir”: “…/dist”
[Build] },
[Build] “include” : [
[Build] “**/*.ts”
[Build] ]
[Build] }
```

But the tsc command cannot access the files even from the workdir:

```
[Build]   [main] Step 9/11 : RUN tsc -p 'src\tsconfig.json'                                  
[Build]   [main]  ---> Running in 819c970a0c46                                               
[Build]   [main] error TS5058: The specified path does not exist: 'src\tsconfig.json'.       
Some services failed to build:                                                               
        main: The command '/bin/sh -c tsc -p 'src\tsconfig.json'' returned a non-zero code: 1
```

Worth noting: the project compiles the TS just fine when just running ‘tsc -p src’ from the terminal of my dev computer and the files are copied correctly when pushing the project to a Balene device NOT running in local mode. I am also able the run the project correctly when pushing from a Mac both to device in and not in local mode.

I am not able to push and build the image on devices in local mode from my Windows computer

**Comment by dutchwyse** [comment]
so any solutions yet ?

**Comment by TobiasEmil** [comment]
Not to my knowledge

**Comment by pdcastro** [comment]
@TobiasEmil (CC: @dutchwyse), thank you for reporting this issue. I have been able to reproduce it. What happens is that the CLI creates a [tar](https://en.wikipedia.org/wiki/Tar_(computing)) stream which should have only forward slashes as the path separator, but under Windows in some cases back slashes are being used instead. There are few github repos that are affected and used by the CLI and the cloud builders. I am working on some pull requests:
* https://github.com/balena-io-modules/node-tar-utils/pull/8
* https://github.com/balena-io-modules/resin-bundle-resolve/pull/36
* https://github.com/balena-io-modules/resin-multibuild/pull/52

They are under review and testing, then they'll be integrated in the CLI and cloud builders. I hope that by the end of next week everything will be tested and deployed/released (production deployment for the cloud builders and a new CLI release).

Thanks again.

**Comment by pdcastro** [comment]
@TobiasEmil, @dutchwyse, I believe this issue was solved in version [10.3.0](https://github.com/balena-io/balena-cli/releases) of the CLI. At least the test case I had now succeeds... Would you mind also testing it to confirm?

**Comment by pdcastro** [comment]
I am closing this issue with the understanding that it was fixed in CLI release [10.3.0](https://github.com/balena-io/balena-cli/releases). If you find that it isn't the case, feel free to re-open it.

**Comment by TobiasEmil** [comment]
Thanks! It work for me now!
